### PR TITLE
Add branch protection and add milestone applier for k/website 1.25

### PR DESCRIPTION
**Pull request by nate-double-u**
Adding branch protection
* release-1.23
* release-1.24

Adding milestone applier
* dev-1.25

/cc @kubernetes/sig-docs-leads 
/sig docs

### CODE REVIEW

File: config/prow/config.yaml [hunk 1]
```diff
@@ -483,6 +483,10 @@ branch-protection:
               protect: true
             release-1.22:
               protect: true
+            release-1.23:
+              protect: true
+            release-1.24:
+              protect: true
```
Comment (reylejano) [inline config/prow/config.yaml:489]:
```suggestion
```
There is no `release-1.24` branch yet, technically `main` is 1.24 at the moment, the branch protection is for the `release-1.23` branch that was created a few days ago

Comment (nate-double-u) [reply config/prow/config.yaml:489]: That makes sense, thanks for that!

**Comment by sftim** [comment]
Looks like this never happened.

**Comment by reylejano** [comment]
closing in favor for https://github.com/kubernetes/test-infra/pull/27251 which applies the branch protection and milestone applier for k/website `dev-1.26`
/close

**Pull request closed without merging**